src/components/vega-lite/index.tsx

```diff
@@ -72,8 +75,15 @@ export class VegaLite extends React.PureComponent<VegaLiteProps, VegaLiteState>
     //     "y": {"field": "b", "type": "quantitative"}
     //   }
     // };
-    const {logger} = this.props;
-    const vlSpec = this.props.spec;
+    const {logger, config} = this.props;
+    let vlSpec = this.props.spec;
+    vlSpec = {
+      ...vlSpec,
+      config: {
+        ...vlSpec.config,
```

Comment:
Wouldn't this override the config users pass to modify vega-lite chart's configuration? My understanding was users pass `vegaliteConfig` to override spec configurations for respective charts (for instance modify the color or axis tick format etc.,)

Reply:
Yes, it will override since we think about the shared config as a theme.  So suppose you have multiple Vega-Lite charts and you have a common config that you pass in.   Each spec can have config that override the common theme (rather than the other way round). :)

Reply:
If that is the case would the common theme ever be applied? Say in the case of bar charts we have defined the fill color to be a blue and if users need to override that would that be possible? I am not sure if I am missing something.

Can you provide an example where a user modifies the fill color of bar charts in voyager with custom config? My assumption was the config user provides are custom config and that would override any default spec config that comes as part of voyager for vega-lite charts.

Reply:
> If that is the case would the common theme ever be applied?

Yes, your common theme may have 10 properties and the spec specific theme may have 1 that override the common theme so 9 properties would still be applied. 

> Can you provide an example where a user modifies the fill color of bar charts in voyager with custom config?

Right now there is no such case. But this is how Vega-Lite treat "common" theme too.  So should CompassQL output some config in the future. 

That said, thinking more about it I think I agree with you that if users explicitly specify the config, it should override what's automatically generated by CompassQL.